utils/asyncForEach.js

```diff
@@ -0,0 +1,7 @@
+const asyncForEach = async (array, callback) => {
+  for (let index = 0; index < array.length; index++) {
+    await callback(array[index], index, array)
+  }
+};
```

Comment:
this is a serial promise resolution pattern, unfortunately it's less colloquial than just writing in the for loop directly (i.e. it's better to just write the loop, there isn't enough benefit in the abstraction). Same for forEach (since neither are functional)